react/components/atoms/switcher/switcher.js

```diff
@@ -0,0 +1,164 @@
+import React, { PureComponent } from "react";
+import { StyleSheet, Text, Animated, View, Platform } from "react-native";
+import PropTypes from "prop-types";
+import { mix } from "yonius";
+
+import { IdentifiableMixin, baseStyles } from "../../../util";
+
+import { Touchable } from "../touchable";
+
+export class Switcher extends mix(PureComponent).with(IdentifiableMixin) {
+    static get propTypes() {
+        return {
+            checked: PropTypes.bool,
+            disabled: PropTypes.bool,
+            checkedText: PropTypes.string,
+            uncheckedText: PropTypes.string,
+            variant: PropTypes.string,
+            animationDuration: PropTypes.number,
+            marginLeftValue: PropTypes.number,
+            colorInputRangeValue: PropTypes.number,
+            onValueUpdate: PropTypes.func
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            checked: false,
+            disabled: false,
+            variant: "colored",
+            checkedText: null,
+            uncheckedText: null,
+            animationDuration: 200,
+            marginLeftValue: 20,
+            colorInputRangeValue: 150,
+            onValueUpdate: () => {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            checkedData: this.props.checked,
+            marginLeft: this.props.checked
+                ? new Animated.Value(this.props.marginLeftValue)
+                : new Animated.Value(0),
+            backgroundColor: this.props.checked
+                ? new Animated.Value(this.props.colorInputRangeValue)
+                : new Animated.Value(0)
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.checked !== this.props.checked) {
+            this.setState({
+                checkedData: this.props.checked,
+                marginLeft: this.props.checked
+                    ? new Animated.Value(this.props.marginLeftValue)
+                    : new Animated.Value(0),
+                backgroundColor: this.props.checked
+                    ? new Animated.Value(this.props.colorInputRangeValue)
+                    : new Animated.Value(0)

```

Comment:
Re-initializaing the values like this seems a bit off. What should be happening is:
1. this.setChecked(this.props.checked)
2. setChecked = what you have right now in onPress

The reason for this is, regardless of the origin of the change in the checked state (user interaction or props change), the way to handle it should be the same, instead of having 2 different strategies.